### PR TITLE
Warning: Invalid argument supplied for foreach() in Model/RecursiveDataProcessor.php on line 28

### PR DESCRIPTION
**Pull request by JMLucas96**
Prevent warning on tfa_ tables.

### CODE REVIEW

File: Console/ReencryptTfaData.php [hunk 1]
```diff
@@ -103,6 +103,11 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 $value = $row[$column];
                 $output->writeln("ciphertext_old: " . $value);
                 $valueDecrypted = $this->encryptor->decrypt($value);
+                if(empty($valueDecrypted)){
+                    $output->writeln("plaintext_old: is null");
+                    $connection->commit();
```
Comment (convenient) [inline Console/ReencryptTfaData.php]: Is the `$connection->commit()` here necessary? should we not be covered by the commit further down?

```php
            $connection->commit();
            $this->cache->clean();
            $output->writeln('Done');
        } catch (\Throwable $throwable) {
```

File: Console/ReencryptTfaData.php [hunk 2]
```diff
@@ -103,6 +103,11 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 $value = $row[$column];
                 $output->writeln("ciphertext_old: " . $value);
                 $valueDecrypted = $this->encryptor->decrypt($value);
+                if(empty($valueDecrypted)){
```
Comment (convenient) [inline Console/ReencryptTfaData.php]:
```suggestion
                if (empty($valueDecrypted)) {
```

**Review by convenient — COMMENTED** [review]
Thank you for your contribution @JMLucas96 

Before I merge this in can I confirm the cause of this failure to decrypt? Is the input `$value` empty ?

**Comment by JMLucas96** [comment]
> Thank you for your contribution @JMLucas96
> 
> Before I merge this in can I confirm the cause of this failure to decrypt? Is the input `$value` empty ?

Hi @convenient , it breaks process when `$this->encryptor->decrypt($value);` returns empty string.
Thank you for your good module, it works pretty well 😄

**Review by convenient — APPROVED** [review]

**Review by cjnewbs — APPROVED** [review]

**Pull request merged**